force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,192 @@
+      been updated
+
+- deleteCloseWonOpportunity(List<Opportunity> oldOpp):
+    - Purpose: When an opportunity is deleted prevent the deletion of a closed won opportunity if the account industry is 'Banking'.
+
+- setPrimaryContact(List<Opportunity> newOpp):
+    - Purpose: When an opportunity is updated set the primary contact on the opportunity to the contact with the title of 'CEO'.
+*/
+public with sharing class OpportunityHelper { 
+    public class CustomOppException extends Exception {}
+
+    public void beforeUpdateValidations(List<Opportunity> newOpp, Map<id, Opportunity> oldOppMap) {
+        for (Opportunity opp : newOpp) {
+            if (opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            } else if (opp.StageName != null && opp.StageName != oldOppMap.get(opp.Id).StageName) { 
+                opp.Description += '\n Stage Change:' + opp.StageName + ':' + DateTime.now().format();
+            } 
+        }
+    }    
+
+    public void deleteCloseWonOpportunity(List<Opportunity> oldOpp) {
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN (SELECT AccountId FROM Opportunity WHERE Id IN :oldOpp)]);
+
+        for (Opportunity opp : oldOpp) {
+            if (opp.StageName == 'Closed Won') {
+                if (accounts.get(opp.AccountId).Industry == 'Banking') {
+                    opp.addError('Cannot delete closed opportunity');
+                }
+            }
+        }
+    }
+
+    public void setType(List<Opportunity> newOpp) {   
+        Opportunity opp = newOpp[0];
+
+        if (opp.Type == null) {
+            opp.Type = 'New Customer';
+        }
+    }
+
+    public void setPrimaryContact(List<Opportunity> newOpp) {
+
+        //Get contacts related to the opportunity account
+        Set<Id> accountIds = new Set<Id>();
+
+        for (Opportunity opp : newOpp) {
+            accountIds.add(opp.AccountId);
+        }
+        
+        Map<Id, Contact> contacts = new Map<Id, Contact>([SELECT Id, FirstName, AccountId FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO' ORDER BY FirstName ASC]);
+        Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+
+        for (Contact cont : contacts.values()) {
+            if (!accountIdToContact.containsKey(cont.AccountId)) {
+                accountIdToContact.put(cont.AccountId, cont);
+            }
+        }
+
+        for (Opportunity opp : newOpp) {
+            if (opp.Primary_Contact__c == null) {
+                if (accountIdToContact.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContact.get(opp.AccountId).Id;
+                }
+            }
+        }
+    }
+
+    public void validateCloseOpportunity(List<Opportunity> oldOpp) {
+        for (Opportunity oppOld : oldOpp) {
+            if (oppOld.IsClosed) {
+                oppOld.addError('Cannot delete closed opportunity');
```

Comment:
Try moving the string inside of the add error to top-level static variables so they are easier to change and understand what they are doing.